service/frontend/dcRedirectionHandler.go

```diff
@@ -548,8 +548,15 @@ func (handler *DCRedirectionHandlerImpl) QueryWorkflow(
 		case targetDC == handler.currentClusterName:
 			resp, err = handler.frontendHandler.QueryWorkflow(ctx, request)
 		default:
-			remoteClient := handler.GetRemoteFrontendClient(targetDC)
-			resp, err = remoteClient.QueryWorkflow(ctx, request)
+			// Only autofoward consistent queries, this is done for two reasons:
+			// 1. Query is meant to be fast, autoforwarding all queries will increase latency.
+			// 2. If eventual consistency was requested then the results from running out of local dc will be fine.
+			if request.GetQueryConsistencyLevel() == shared.QueryConsistencyLevelStrong {
+				remoteClient := handler.GetRemoteFrontendClient(targetDC)
+				resp, err = remoteClient.QueryWorkflow(ctx, request)
+			} else {
+				resp, err = handler.frontendHandler.QueryWorkflow(ctx, request)
+			}
```

Comment:
Is there a unit test for this?

Reply:
Yes service/frontend/dcRedirectionHandler_test.go